tests/base.py

```diff
@@ -0,0 +1,196 @@
+
+    def expected_metadata(self):
+        return {
+            "campaign": {
+                self.PRIMARY_KEYS: {"id"},
+                self.REPLICATION_METHOD: self.FULL_TABLE
+            },
+            "content_area":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "data_extension.test emails":{
+                self.PRIMARY_KEYS: {"_CustomObjectKey", "ID"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+            },
+            "data_extension.This is a test":{
+                self.PRIMARY_KEYS: {"_CustomObjectKey", "ID"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+            },
+            "data_extension.my_test":{
+                self.PRIMARY_KEYS: {"_CustomObjectKey", "ID"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+            },
+            "data_extension.test 1":{
+                self.PRIMARY_KEYS: {"_CustomObjectKey", "ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"JoinDate"},
+            },
+            "email":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "event": {
+                self.PRIMARY_KEYS: {"SendID", "EventType", "SubscriberKey", "EventDate"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"EventDate"},
+            },
+            "folder":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "list":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "list_send":{
+                self.PRIMARY_KEYS: {"ListID", "SendID"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
+            },
+            "list_subscriber":{
+                self.PRIMARY_KEYS: {"SubscriberKey", "ListID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "send":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            },
+            "subscriber":{
+                self.PRIMARY_KEYS: {"ID"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"ModifiedDate"},
+            } 
+        }
+
+    def streams_to_select(self):
+        return set(self.expected_metadata().keys()) - {'event', 'list_subscriber', 'subscriber'}
```

Comment:
@harshpatel4crest Please add a comment to explain the reason for skipping these streams

Reply:
Done